lib/SIL/ValueOwnership.cpp

```diff
@@ -260,6 +253,16 @@ FORWARDING_OWNERSHIP_INST(Upcast)
 FORWARDING_OWNERSHIP_INST(UncheckedEnumData)
 FORWARDING_OWNERSHIP_INST(SelectEnum)
 FORWARDING_OWNERSHIP_INST(Enum)
+// NOTE: init_existential_ref from a reference counting perspective is not
+// considered to be "owned" since it doesn't affect reference counts. That being
+// said in the past, we wanted to conceptually treat it as an owned value that
+// produces owned things, rather than a forwarding thing since initialization is
+// generally a consuming operation. That being said, there are often cases in
+// class based code where we are propagating around a plus zero version of a
+// value and need to wrap the class in an existential wrapper in an intermediate
+// frame from usage. In such cases, we have been creating unnecessary ref count
+// traffic in code.
```

Comment:
@gottesmm it's hard to understand this comment because you're explaining a previous implementation. Just explain the key points of the current implementation:

- it has no net effect on RC
- it consumes an owned value to produce an owned existential
- or it "extends" a borrowed value to produce a borrowed existential inside the lifetime of the original value

I'm saying the a forwarding cast "extends" a borrow, while a guaranteed phi (or tuple/struct) "reborrows" the value, unless you have other terms

Reply:
I am fine with that.